### ISSUE TITLE
[Bug]: Narrator doesn't read ComboBox options correctly on arrow key

### ISSUE THREAD

**Issue by alvalencia-ms** [microsoft/fluentui#25124]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: Linux 5.4 Ubuntu 22.04.1 LTS 22.04.1 LTS (Jammy Jellyfish)
    CPU: (32) x64 Intel(R) Xeon(R) Platinum 8272CL CPU @ 2.60GHz
    Memory: 4.62 GB / 62.81 GB
    Container: Yes
    Shell: 5.1.16 - /bin/bash

Package Versions:
    @fluentui/react: ^8.91.1
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://codepen.io/alvalencia/pen/jOxvaLm

## Actual Behavior
When changing options in the combobox using [shift + arrow] narrator reads:
    If the previous option is longer, it will read "[space] unselected"
    If the previous option is shorter, it will read "[difference between the two texts] selected"
        - e.g. if previous option was Florida, and next option is Washington, it will read "ton selected"

## Expected Behavior
Narrator should read the entire option text

---

An issue was already closed due to inactivity and unable to repro. https://github.com/microsoft/fluentui/issues/21135

### Requested priority

High

### Products/sites affected

OWA

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by miroslavstastny** [comment]
Thanks for the codesandbox, I am able to repro.

**Comment by JustSlone** [comment]
@alvalencia-ms can you file this issue internally and link to this github issue? https://aka.ms/fluentui-a11y. This will help track the issue.
